examples/core/dof/app.js

```diff
@@ -0,0 +1,433 @@
+
+      const instancedBoxes = new Model(gl, { 
+        program: sceneProgram,
+        attributes: {
+          positions: positionBuffer,
+          normals: normalBuffer,
+          uvs: uvBuffer,
+
+          // Attributes are limited to 4 components,
+          // So we have to split the matrices across
+          // 4 attributes. They're reconstructed in 
+          // the vertex shader.
+          modelMatCol1: {
+            buffer: matrixBuffer,
+            size: 4,
+            stride: 64,
+            offset: 0,
+            divisor: 1
+          },
+          modelMatCol2: {
+            buffer: matrixBuffer,
+            size: 4,
+            stride: 64,
+            offset: 16,
+            divisor: 1
+          },
+          modelMatCol3: {
+            buffer: matrixBuffer,
+            size: 4,
+            stride: 64,
+            offset: 32,
+            divisor: 1
+          },
+          modelMatCol4: {
+            buffer: matrixBuffer,
+            size: 4,
+            stride: 64,
+            offset: 48,
+            divisor: 1
+          },
+          indices: indexBuffer
+        },
+        uniforms: {
+          uTexture: textures[0]
+        },
+        instanced: true,
+        instanceCount: NUM_BOXES,
+        drawMode: GL.TRIANGLES,
+        vertexCount: cubeGeo.indices.value.length
+      });
+
+      const quadVertexArray = new VertexArray(gl, {
+        program: dofProgram,
+        attributes: {
+          aPosition: new Buffer(gl, new Float32Array(QUAD_VERTS))
+        }
+      });
+
+      return {
+        instancedBoxes,
+        boxXforms,
+        boxMatrices,
+        matrixBuffer,
+        mainFramebuffer,
+        dofFramebuffer,
+        quadVertexArray,
+        dofProgram
+      };
+    });
+  },
+
+  onRender: ({gl, tick, width, height, aspect, instancedBoxes, boxMatrices, boxXforms, matrixBuffer, mainFramebuffer, dofFramebuffer, quadVertexArray, dofProgram}) => {
+
+    mainFramebuffer.resize(gl.drawingBufferWidth, gl.drawingBufferHeight);
+    dofFramebuffer.resize(gl.drawingBufferWidth, gl.drawingBufferHeight);
+
+    clear(gl, {color: [0, 0, 0, 1], depth: true, framebuffer: mainFramebuffer});
+
+    const camView = new Matrix4().lookAt({eye: [3, 1.5, 3], center: [0, 0, 0], up: [0, 1, 0]});
+    const camProj = new Matrix4().perspective({fov: radians(75), aspect, near: 0.1, far: 30});
+
+    ////////////////////////////////////////
+    // Update model matrix data and then
+    // update the attribute buffer
+    ////////////////////////////////////////
+
+    for (let i = 0; i < NUM_BOXES; ++i) {
+      let box = boxXforms[i];
+      box.rotate[0] += 0.01;
+      box.rotate[1] += 0.02;
+      box.matrix.identity().translate(box.translate).rotateXYZ(box.rotate).scale(box.scale);
+      boxMatrices.set(box.matrix, i * 16);
+    }
+
+    matrixBuffer.setData(boxMatrices);
+
+    ////////////////////////////////////
+    // Draw boxes to main framebuffer
```

Comment:
Nit: `main framebuffer` comments is confusing, you are actully rendering to offscreen frame buffer here, so may be change this comment and names to `offscreenFramebuffer` or `sceneFramebuffer`.  `dofProgram` is actually rendering to main framebuffer.